pkg/api/openshiftcluster.go

```diff
@@ -32,6 +35,29 @@ func (oc *OpenShiftCluster) UsesWorkloadIdentity() bool {
 	return oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil
 }
 
+// ClusterMsiResourceId returns the resource ID of the cluster MSI or an error
+// if it encounters an issue while grabbing the resource ID from the cluster
+// doc. It is written under the assumption that there is only one cluster MSI
+// and will have to be refactored if we ever use more than one.
+func (oc *OpenShiftCluster) ClusterMsiResourceId() (*arm.ResourceID, error) {
+	var clusterMsi *arm.ResourceID
+	if oc.Identity != nil && oc.Identity.UserAssignedIdentities != nil && len(oc.Identity.UserAssignedIdentities) == 1 {
+		for msiResourceId := range oc.Identity.UserAssignedIdentities {
+			var err error
+			clusterMsi, err = arm.ParseResourceID(msiResourceId)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	if clusterMsi == nil {
+		return nil, errors.New("could not find cluster MSI in cluster doc or unexpectedly found more than one cluster MSI")
```

Comment:
Would it make sense to separate out these cases in code (and in your error message)?
1) Could not find cluster MSI in cluster doc
2) Unexpectedly found more than one cluster MSI

Reply:
Agreed and done 👍🏻